graylog-storage-opensearch3/src/main/java/org/graylog/storage/opensearch3/OfficialOpensearchClientProvider.java

```diff
@@ -94,31 +98,59 @@ private static OfficialOpensearchClient createClient(List<URI> uris, IndexerJwtA
         }
 
         final ApacheHttpClient5TransportBuilder builder = ApacheHttpClient5TransportBuilder.builder(hosts);
+
+        builder.setRequestConfigCallback(requestConfigBuilder ->
+                requestConfigBuilder
+                        .setConnectionRequestTimeout(timeout(clientConfiguration.elasticsearchConnectTimeout()))
+                        .setResponseTimeout(timeout(clientConfiguration.elasticsearchSocketTimeout()))
+                        .setExpectContinueEnabled(clientConfiguration.useExpectContinue())
+                        .setAuthenticationEnabled(true)
+        );
+
+        builder.setChunkedEnabled(clientConfiguration.compressionEnabled());
+
         builder.setHttpClientConfigCallback(httpClientBuilder -> {
+
             final TlsStrategy tlsStrategy = ClientTlsStrategyBuilder.create()
                     .setSslContext(sslcontext)
                     // See https://issues.apache.org/jira/browse/HTTPCLIENT-2219
                     .setTlsDetailsFactory(sslEngine -> new TlsDetails(sslEngine.getSession(), sslEngine.getApplicationProtocol()))
                     .build();
 
-            final PoolingAsyncClientConnectionManager connectionManager = PoolingAsyncClientConnectionManagerBuilder
-                    .create()
+            httpClientBuilder.setConnectionManager(PoolingAsyncClientConnectionManagerBuilder.create()
+                    .setMaxConnPerRoute(clientConfiguration.elasticsearchMaxTotalConnectionsPerRoute())
+                    .setMaxConnTotal(clientConfiguration.elasticsearchMaxTotalConnections())
+                    .setDefaultConnectionConfig(
+                            ConnectionConfig.custom()
+                                    .setConnectTimeout(timeout(clientConfiguration.elasticsearchConnectTimeout()))
+                                    .setSocketTimeout(timeout(clientConfiguration.elasticsearchSocketTimeout()))
+                                    .build()
+
+                    )
                     .setTlsStrategy(tlsStrategy)
-                    .build();
+                    .build());
+
+            httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
 
             if (indexerJwtAuthToken.isJwtAuthEnabled()) {
                 httpClientBuilder.addRequestInterceptorLast(jwtInterceptor(indexerJwtAuthToken));
             }
 
-            return httpClientBuilder
-                    .setDefaultCredentialsProvider(credentialsProvider)
-                    .setConnectionManager(connectionManager);
+            if (clientConfiguration.muteDeprecationWarnings()) {
```

Comment:
Shouldn't be `if (false == clientConfiguration.muteDeprecationWarnings()) ...{` ?